packages/o/opentelemetry-cpp/xmake.lua

```diff
@@ -0,0 +1,115 @@
+        "opentelemetry_exporter_in_memory",
+        "opentelemetry_http_client_curl",
+        "opentelemetry_proto_grpc",
+        "opentelemetry_otlp_recordable",
+        "opentelemetry_proto",
+        "opentelemetry_logs",
+        "opentelemetry_trace",
+        "opentelemetry_metrics",
+        "prometheus-cpp-pull",
+        "prometheus-cpp-push",
+        "prometheus-cpp-core",
+        "opentelemetry_resources",
+        "opentelemetry_common",
+        "opentelemetry_version",
+        "opentelemetry_cpp"
+    )
+
+    on_load(function (package)
+        if package:config("otlpGrpc") then
+            package:add("deps", "grpc")
+        elseif package:config("otlpHttp") or package:config("otlpFile") then
+            package:add("deps", "protobuf-cpp")
+        elseif package:config("abseil") then
+            package:add("deps", "abseil")
+        end
+        if package:config("shared") and package:is_plat("windows") then
+            package:add("defines", "OPENTELEMETRY_BUILD_IMPORT_DLL")
+        end
+    end)
+
+    on_install("!iphoneos and !cross|!aarch64", function (package)
+        local configs = {
+            "-DBUILD_TESTING=OFF",
+            "-DWITH_EXAMPLES=OFF",
+            "-DWITH_BENCHMARK=OFF",
+            "-DWITH_FUNC_TESTS=OFF",
+            "-DCMAKE_CXX_STANDARD=14"
+        }
+        if package:has_tool("cxx", "clang", "clangxx") then
+            package:add("cxxflags", "clang::-Wno-missing-template-arg-list-after-template-kw")
+            table.insert(configs, "-DCMAKE_CXX_FLAGS=-Wno-missing-template-arg-list-after-template-kw")
+        end
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DCMAKE_POSITION_INDEPENDENT_CODE=" .. (package:config("pic") and "ON" or "OFF"))
+        if package:config("shared") then
+            if package:is_plat("windows") then
+                table.insert(configs, "-DOPENTELEMETRY_BUILD_DLL=ON")
+            else
+                table.insert(configs, "-DBUILD_SHARED_LIBS=ON")
+            end
+        else
+            table.insert(configs, "-DBUILD_SHARED_LIBS=OFF")
+        end
+        
+        table.insert(configs, "-DWITH_OTLP_GRPC=" .. (package:config("otlpGrpc") and "ON" or "OFF"))
+        table.insert(configs, "-DWITH_OTLP_HTTP=" .. (package:config("otlpHttp") and "ON" or "OFF"))
+        table.insert(configs, "-DWITH_OTLP_FILE=" .. (package:config("otlpFile") and "ON" or "OFF"))
+        table.insert(configs, "-DWITH_PROMETHEUS=" .. (package:config("prometheus") and "ON" or "OFF"))
+        table.insert(configs, "-DWITH_API_ONLY=" .. (package:config("apiOnly") and "ON" or "OFF"))
+        local stl = package:config("stl")
+        if stl == "true" then
+            stl = "on"
+        elseif stl == "false" then
+            stl = "off"
+        end
+        stl = string.upper(stl)
+        table.insert(configs, "-DWITH_STL=" .. stl)
```

Comment:
why not just `table.insert(configs, "-DWITH_STL=" .. (package:config("stl") and "ON" or "OFF"))`?